blog/2021/12/dtable-performance-assessment.md

```diff
@@ -0,0 +1,261 @@
+
+Contrary to the standard reduction benchmarks, the `DTable` doesn't offer a speedup compared to `DataFrames.jl` across all the data sizes.
+It looks like the current algorithm has a significant overhead that can be observed as a lower bound to the performance at smaller data sizes.
+For the benchmarks with the smaller `unique_values` count, the `DTable` manages to catch up to `DataFrames.jl` at bigger data sizes.
+This may indicate that by increasing the data size further, we might eventually reach a point where the `DTable` provides a performance improvement over `DataFrames.jl` in this scenario.
+
+DTable command: `r = reduce(fit!, g, cols=[:a2], init=Mean())`
+
+![](/assets/blog/2021-dtable/grouped_reduce_mean_singlecol.svg)
+
+
+## Grouped reduction (all columns)
+
+The results for the all-columns reduction look very similiar to single-column.
+The `DTable` managed to offer an average ~22.3 times speeup over `Dask`.
+
+Again, the `DTable` is heavily falling behind `DataFrames.jl` on smaller data sizes due to the significant entry overhead acting as a lower performance bound at smaller data sizes.
+
+DTable command: `r = reduce(fit!, g, init=Mean())`
+
+![](/assets/blog/2021-dtable/grouped_reduce_mean_allcols.svg)
+
+# Implementation details (for interested users)
+
+The `DTable` is built on top of `Dagger` and `Tables.jl`, and currently resides within the `Dagger.jl` package.
+That means it can run in any environment `Dagger` is capable of running in.
+You should be able to use the `DTable` effectively on your local machine in a threaded environment, on a bigger machine with many workers and threads, or have the workload spread around multiple machines and workers in your cluster.
+
+The `DTable` uses `Dagger`'s new "Eager API", which means that all the parallelized calls are executed using `Dagger.spawn` and `Dagger.@spawn`.
+Memory is managed by `Dagger` through the usage of `MemPool.jl`.
+Upgrades to the related projects in the future will hopefully yield performance and functionality improvements for the `DTable`.
+
+
+Because of the dependencies of the `DTable` on other projects, its focus is completely on delivering `Tables.jl` compatible algorithms and interfaces to address the growing needs for processing big tabular data.
+
+We hope that the `Tables.jl` interface will grow to include an even wider range of functionality while still providing great intercompatibility with other Julia packages.
+
+For more details, please visit the [Dagger documentation](https://juliaparallel.github.io/Dagger.jl/dev/).
+
+# Some caveats
+
+There are some pending PRs that haven't been merged into Julia yet that improve the thread safety of `Distributed`, which directly affects `Dagger.jl` stability. The user experience may occasionally be interrupted when extensively using the `DTable` in a threaded or mixed environment by occasional hangs or crashes.
+
+We hope to include all the necessary fixes in future patches to Julia 1.7.
```

Comment:
Perhaps this should be Julia 1.8 (since 1.7 is released)?

Reply:
The 2 PRs, which are relevant here, are marked for backport into 1.7, so they will eventually be there in 1.7.x
I can change it to something more generic like "future Julia versions"